client/sections.js

```diff
@@ -134,36 +134,6 @@ const sections = [
 		module: 'calypso/my-sites/posts',
 		group: 'sites',
 	},
-	{
```

Comment:
These paths have been moved directly in the site-settings module in order to avoid triggering the section-middleware twice. The main issue was that the generic /settings path was converted to a regex that conflicted with the other settings specific pages (e.g. settings/writing). 

When that happened, the sections-middleware was called twice when the navigation was opened. Because of this, the navigation toggled (with an redux action, as it should) and immediately un-toggled (because the middleware was fired a second time, which in turned triggered the redux action once again).

However, while this implementation works on my environment, my main concern is if there are any sizeable performance effects we should be aware of because we've moved these paths into the site-settings module, instead of loading them asynchronously individually (from my understanding, that's what we did before this change).

Reply:
The changes test well for me but I am worried of the changes we introduce here. 

> The main issue was that the generic /settings path was converted to a regex that conflicted with the other settings specific pages (e.g. settings/writing).

Could we maybe fix this which seems like the root of the problem?

Reply:
Unfortunately I think that would be higher impact change since the regex pattern is automatically generated by a [generic helper function](https://github.com/Automattic/wp-calypso/blob/trunk/client/utils.js) that's used on all paths in Calypso. I think in theory we could make something more configurable, but that would be a bit out of scope and personally I think would be a bit risky. 

From a functional perspective, in the worse case scenario, this approach would affect only the routes we've extracted, which is are known.

Reply:
The problem seems to be that
`activateSection()` runs twice
https://github.com/Automattic/wp-calypso/blob/ed3e91545e431da92beeffcbe80289eadde067c0/client/sections-middleware.js#L24

because this regex
`return new RegExp( '^' + path + '(/.*)?$' );`
https://github.com/Automattic/wp-calypso/blob/60069449be91a6618d3d250fa9ccff8be2248b31/client/utils.js#L8 

makes eg `/settings/discussion` also match `/settings`. 

The problem didn't appear in the past cause this menus were in a secondary navigation so accessing them wouldn't require to open the sidebar. We have removed the secondary navigation here https://github.com/Automattic/wp-calypso/pull/53824

Reply:
I probably echo your discoveries, disregard in that case. I am just keeping notes.

Reply:
That's correct, thanks for the additional comment. 
> The problem didn't appear in the past cause these menus were in a secondary navigation so accessing them wouldn't require to open the sidebar

I don't think the change was introduced in there, I think it was present before this was introduced (that PR doesn't include any changes on how we do the routing). I think before the change the issue wasn't as visible as is now, since the menus were in a subsection (and possible reproducing only refreshing the page?).

Also, the issue was also present in Posts > Categories and Posts > Tags.

Reply:
Switching the `'/settings'` path to `'/settings/general'` in `client/sections.js` works well for me regarding the mobile menu opening. And anyway `settings` did redirect to `settings/general` already.

BUT, whenever I open the menu I get redirected to the initial screen I booted the app. I suppose this is fixed with the part 2 of this fix?

Reply:
> Switching the '/settings' path to '/settings/general' in client/sections.js works well for me regarding the mobile menu opening. And anyway settings did redirect to settings/general already.

That's what we initially did, but Miguel pointed out that doing so we'll break the current behaviours:
- A user going to wordpress.com/settings/ (note the missing domain) will redirect the user to the site selection component where they'll choose which site they'll be redirected to.
- A user going to wordpress.com/settings/writing (or any other submenu like security, jetpack, performance, etc) will also have to be redirected to the site selection component.

>  BUT, whenever I open the menu I get redirected to the initial screen I booted the app. I suppose this is fixed with the part 2 of this fix?

Yup, that's correct.

Reply:
> A user going to wordpress.com/settings/ (note the missing domain) will redirect the user to the site selection component where they'll choose which site they'll be redirected to.

This would need to be `wordpress.com/settings/general` or we can add a redirect.

> This is fixed by doing the same as `/settings/podcasting/:site_id` where a `/settings/podcasting` is also declared here 

https://github.com/Automattic/wp-calypso/blob/28810f1dd55bb7cb25b38cebf3cfb1488e580b04/client/my-sites/site-settings/settings-writing/index.js#L39-L49

The way that currently actually works in production is very interesting, I would say that it works by accident due to that regex.

Removing those `/settings/*` sections would still trigger the `activateSection()` for `/settings` which is something we wouldn't want anyway right?

I may be wrong though, and all the above was intended 😄!